### PR TITLE
Handle Personas in the search/detail API (fix #2660)

### LINKED ISSUE #2660: Handle Personas in the search/detail API
We don't currently return Personas attributes in the search/detail API, making them a bit useless. This will require a bit of thinking.

A couple thoughts about the code:
- Do we add a different serializer and switch according to the Addon `type` ? Or do we add somehow the extra fields in the regular `AddonSerializer`/`ESAddonSerializer` ?
- In any case, it means we need to add the fields to ES, and then we need to make sure to create a fake `Persona` instance in the serializer as well, otherwise we'll cause a bunch of SQL queries in the search API

### PR DESCRIPTION
**Pull request by diox**

### CODE REVIEW

File: src/olympia/addons/serializers.py [hunk 1]
```diff
@@ -114,6 +121,20 @@ def get_review_url(self, obj):
     def get_icon_url(self, obj):
         return absolutify(obj.get_icon_url(64))
 
+    def get_theme_data(self, obj):
+        theme_data = None
+
+        if obj.type == amo.ADDON_PERSONA:
+            try:
+                if obj.persona is None or hasattr(obj.persona, '_broken'):
```
Comment (EnTeQuAk) [inline src/olympia/addons/serializers.py]: I think you should comment that like you did below, either with the django ticket or with a reference to the comment below.

File: src/olympia/addons/tests/test_indexers.py [hunk 2]
```diff
@@ -167,26 +181,57 @@ def test_extract_translations(self):
...
+        self.addon = addon_factory(persona_id=42, type=amo.ADDON_PERSONA)
+        persona = self.addon.persona
+        persona.__dict__.update({
+            'header': u'myheader.jpg',
+            'footer': u'myfooter.jpg',
+            'accentcolor': u'336699',
+            'textcolor': u'f0f0f0',
+            'author': u'Me-me-me-Myself',
+            'display_username': u'my-username',
+        })
```
Comment (EnTeQuAk) [inline src/olympia/addons/tests/test_indexers.py]: I don't see why we can't do

```python
persona.header = u'myheader.jpg'
persona.footer = u'myfooter.jpg'
persona.accentcolor = u'336699'
persona.textcolor = u'f0f0f0'
persona.author = u'Me-me-me-Myself'
persona.display_username = u'my-username'
```

File: src/olympia/addons/tests/test_serializers.py [hunk 3]
```diff
@@ -152,6 +153,51 @@ def test_translations(self):
...
+        self.addon = addon_factory(persona_id=42, type=amo.ADDON_PERSONA)
+        persona = self.addon.persona
+        persona.__dict__.update({
+            'header': u'myheader.jpg',
+            'footer': u'myfooter.jpg',
+            'accentcolor': u'336699',
+            'textcolor': u'f0f0f0',
+            'author': u'Me-me-me-Myself',
+            'display_username': u'my-username',
+        })
```
Comment (EnTeQuAk) [inline src/olympia/addons/tests/test_serializers.py]: same here, (why use ``__dict__``?)

File: src/olympia/addons/tests/test_serializers.py [hunk 4]
```diff
@@ -152,6 +153,51 @@ def test_translations(self):
...
+            description=u'<script>alert(42)</script>My Personä description',
+            type=amo.ADDON_PERSONA)
+        persona = self.addon.persona
+        persona.__dict__.update({
+            'header': u'myheader.jpg',
+            'footer': u'myfooter.jpg',
+            'accentcolor': u'336699',
+            'textcolor': u'f0f0f0',
+            'author': u'Me-me-me-Myself',
+            'display_username': u'my-username',
```
Comment (EnTeQuAk) [inline src/olympia/addons/tests/test_serializers.py]: same here, (why use ``__dict__``?)

File: src/olympia/addons/tests/test_serializers.py [hunk 5]
```diff
@@ -152,6 +153,51 @@ def test_translations(self):
...
+        })
+        persona.save()
+        result = self.serialize()
+        assert result['theme_data'] == persona.theme_data
+        assert '<script>' not in result['theme_data']['description']
+        assert '&lt;script&gt;' in result['theme_data']['description']
+
+    def xxtest_handle_persona_without_persona_data(self):
```
Comment (EnTeQuAk) [inline src/olympia/addons/tests/test_serializers.py]: Is this supposed to be skipped?

File: src/olympia/addons/tests/test_serializers.py [hunk 6]
```diff
@@ -203,3 +249,28 @@ def test_icon_url_without_modified_date(self):
...
+        lands, it will need a reindex first."""
+        self.addon = addon_factory(type=amo.ADDON_PERSONA)
+        persona = self.addon.persona
+        persona.__dict__.update({
+            'header': u'myheader.jpg',
+            'footer': u'myfooter.jpg',
+            'accentcolor': u'336699',
+            'textcolor': u'f0f0f0',
+            'author': u'Me-me-me-Myself',
+            'display_username': u'my-username',
```
Comment (EnTeQuAk) [inline src/olympia/addons/tests/test_serializers.py]: same here, (why use ``__dict__``?)

File: src/olympia/api/fields.py [hunk 7]
```diff
@@ -73,8 +74,8 @@ def __init__(self, *args, **kwargs):
     def fetch_all_translations(self, obj, source, field):
         translations = field.__class__.objects.filter(
             id=field.id, localized_string__isnull=False)
-        return dict((to_language(trans.locale), unicode(trans))
-                    for trans in translations) if translations else None
+        return {to_language(trans.locale): unicode(trans)
+                for trans in translations} if translations else None
```
Comment (EnTeQuAk) [inline src/olympia/api/fields.py]: :+1:

**Comment by EnTeQuAk** [comment]
r+wc

File: src/olympia/addons/tests/test_serializers.py (hunk 5 above)
Comment (diox) [inline src/olympia/addons/tests/test_serializers.py]: no, good catch

File: src/olympia/addons/tests/test_indexers.py (hunk 2 above)
Comment (diox) [inline src/olympia/addons/tests/test_indexers.py]: Yeah I'll do that.

**Pull request merged**